src/core/pointcloud/qgsremoteeptpointcloudindex.cpp

```diff
@@ -0,0 +1,291 @@
+{
+  mTileDownloadManager = QgsApplication::tileDownloadManager();
+}
+
+QgsRemoteEptPointCloudIndex::~QgsRemoteEptPointCloudIndex() = default;
+
+QList<IndexedPointCloudNode> QgsRemoteEptPointCloudIndex::nodeChildren( const IndexedPointCloudNode &n ) const
+{
+  // Potential bug: what if loadNodeHierarchy fails and returns false
+  Q_ASSERT( loadNodeHierarchy( n ) );
+
+  QList<IndexedPointCloudNode> lst;
+  int d = n.d() + 1;
+  int x = n.x() * 2;
+  int y = n.y() * 2;
+  int z = n.z() * 2;
+
+  for ( int i = 0; i < 8; ++i )
+  {
+    int dx = i & 1, dy = !!( i & 2 ), dz = !!( i & 4 );
+    IndexedPointCloudNode n2( d, x + dx, y + dy, z + dz );
+    if ( mHierarchy.contains( n2 ) )
+      lst.append( n2 );
+  }
+  return lst;
+}
+
+void QgsRemoteEptPointCloudIndex::load( const QString &url )
+{
+  mUrl = QUrl( url );
+
+  QStringList splitUrl = url.split( '/' );
+
+  mUrlFileNamePart = splitUrl.back();
+  splitUrl.pop_back();
+  mUrlDirectoryPart = splitUrl.join( '/' );
+
+  QNetworkRequest nr( url );
+
+  QgsBlockingNetworkRequest req;
+  QgsBlockingNetworkRequest::ErrorCode errCode = req.get( nr );
+  if ( errCode != QgsBlockingNetworkRequest::NoError )
+  {
+    QgsDebugMsg( QStringLiteral( "Request failed: " ) + url );
+    mIsValid = false;
+    return;
+  }
+
+  QgsNetworkReplyContent reply = req.reply();
+  bool success = loadSchema( reply.content() );
+
+  if ( success )
+  {
+    // try to import the metadata too!
+    QNetworkRequest nr( QStringLiteral( "%1/ept-sources/manifest.json" ).arg( mUrlDirectoryPart ) );
+
+    QgsBlockingNetworkRequest req;
+    QgsBlockingNetworkRequest::ErrorCode errCode = req.get( nr );
+    if ( errCode == QgsBlockingNetworkRequest::NoError )
+    {
+      QgsNetworkReplyContent reply = req.reply();
+      const QByteArray manifestJson = reply.content();
+      loadManifest( manifestJson );
+    }
+  }
+
+  mIsValid = success;
+}
+
+QgsPointCloudBlock *QgsRemoteEptPointCloudIndex::nodeData( const IndexedPointCloudNode &n, const QgsPointCloudRequest &request )
+{
+  if ( !loadNodeHierarchy( n ) )
+    return nullptr;
+
+  QString fileUrl;
+  QString fileName;
+  if ( mDataType == QLatin1String( "binary" ) )
+  {
+
+    fileUrl = QStringLiteral( "%1/ept-data/%2.bin" ).arg( mUrlDirectoryPart, n.toString() );
+    fileName = QStringLiteral( "/tmp/%1.bin" ).arg( n.toString() );
+  }
+  else if ( mDataType == QLatin1String( "zstandard" ) )
+  {
+    fileUrl = QStringLiteral( "%1/ept-data/%2.zst" ).arg( mUrlDirectoryPart, n.toString() );
+    fileName = QStringLiteral( "/tmp/%1.zst" ).arg( n.toString() );
+  }
+  else if ( mDataType == QLatin1String( "laszip" ) )
+  {
+    fileUrl = QStringLiteral( "%1/ept-data/%2.laz" ).arg( mUrlDirectoryPart, n.toString() );
+    fileName = QStringLiteral( "/tmp/%1.laz" ).arg( n.toString() );
+  }
+  else
+  {
+    return nullptr;  // unsupported
+  }
+
+  QgsFileDownloader downloader( fileUrl, fileName );
```

Comment:
This is because if I use QgsTileDownloadManager I need to redirect the downloaded data to a file, so here I just download the file directly from fileUrl to fileName path (skips the additional explicit writing to the file).